The EL patchset modifies the way defines are handled during the build.

For example, building with the vanilla Gentoo ebuild would lead to a
compile line similar to this:

libtool: compile:  i686-pc-linux-gnu-gcc -DPACKAGE_NAME=\"libnfsidmap\" \
 -DPACKAGE_TARNAME=\"libnfsidmap\" -DPACKAGE_VERSION=\"0.24\" \
 "-DPACKAGE_STRING=\"libnfsidmap 0.24\"" \
 -DPACKAGE_BUGREPORT=\"linux-nfs@vger.kernel.org\" -DPACKAGE_URL=\"\" \
 -DPACKAGE=\"libnfsidmap\" -DVERSION=\"0.24\" -DSTDC_HEADERS=1 \
 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 \
 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 \
 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" \
 -DENABLE_LDAP=1 -DSTDC_HEADERS=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 \
 -DHAVE_UNISTD_H=1 -DHAVE_ERRNO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_MALLOC=1 \
 -DHAVE_STRCHR=1 -DHAVE_STRDUP=1 -DHAVE_GETGROUPLIST=1 \
 -I. -O2 -march=i686 -pipe -c nss.c  -fPIC -DPIC -o .libs/nss.o

The EL patches roll all these defines up into a config.h file that must
be included.

After this modification, a compile line will look similar to this:

libtool: compile:  i686-pc-linux-gnu-gcc -DHAVE_CONFIG_H \
 -I. -O2 -march=i686 -pipe -c nss.c  -fPIC -DPIC -o .libs/nss.o

diff -Naur libnfsidmap-0.24.a/getgrouplist.c libnfsidmap-0.24.b/getgrouplist.c
--- libnfsidmap-0.24.a/getgrouplist.c	2012-05-11 11:05:40.170262009 -0500
+++ libnfsidmap-0.24.b/getgrouplist.c	2012-05-11 11:10:59.503527779 -0500
@@ -32,6 +32,10 @@
  * SUCH DAMAGE.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/types.h>
 #include <grp.h>
 #include <string.h>
